### PR TITLE
Fetch JWT SVID workload API implementation

### PR DESCRIPTION
**Pull request by glaucimar**
- created a new file for jwt related methods tests
- added implementation for fetch_jwt_svid method
- added tests for fetch_jwt_svid
- created FetchJwtSvidError exception

This partially addresses #62.

### CODE REVIEW

File: src/pyspiffe/workloadapi/default_workload_api_client.py [hunk 1]
```diff
@@ -134,19 +139,28 @@ def fetch_x509_bundles(self) -> X509BundleSet:
         return self._create_bundle_set(response.bundles)
 
     def fetch_jwt_svid(
-        self, audiences: Set[str], subject: Optional[str] = None
+        self, audiences: List[str], subject: Optional[str] = None
```
[the remark this answers is not in the document]
Comment (maxlambrecht) [reply src/pyspiffe/workloadapi/default_workload_api_client.py:142]: The subject should be of type `SpiffeId`

File: src/pyspiffe/workloadapi/default_workload_api_client.py [hunk 2]
```diff
@@ -239,3 +253,21 @@ def _create_x509_bundle(trust_domain: TrustDomain, bundle: Any) -> X509Bundle:
...
         except Exception as e:
             raise FetchX509BundleError(normalized_exception_message(e))
+
+    def _call_fetch_jwt_svids(
+        self, audience: List[str], spiffe_id: Optional[str] = None
+    ) -> workload_pb2.JWTSVIDResponse:
+
+        try:
+            request = workload_pb2.JWTSVIDRequest()
+            for aud in audience:
```
Comment (maxlambrecht) [inline src/pyspiffe/workloadapi/default_workload_api_client.py:263]: This `for` could be replaced by `request.audience.extend(audience)`

File: src/pyspiffe/workloadapi/workload_api_client.py [hunk 3]
```diff
@@ -48,13 +51,13 @@ def fetch_x509_bundles(self) -> X509BundleSet:
 
     @abstractmethod
     def fetch_jwt_svid(
-        self, audiences: Set[str], subject: Optional[str] = None
+        self, audiences: List[str], subject: Optional[str] = None
```
Comment (maxlambrecht) [inline src/pyspiffe/workloadapi/workload_api_client.py:54]: Same comments about `audiences` as a `Set` and `subject` as `SpiffeId`

**Review by maxlambrecht — COMMENTED** [review]
Thanks @glaucimar for this!

File: src/pyspiffe/workloadapi/default_workload_api_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (glaucimar) [reply src/pyspiffe/workloadapi/default_workload_api_client.py]: > Why use a list, can the audiences be duplicated ?

I used List as this was used in JwtSvid already... but as Set is more performatic, I made the change.



> The subject should be of type `SpiffeId`

Ok, done.

File: src/pyspiffe/workloadapi/default_workload_api_client.py [hunk 4]
```diff
@@ -139,24 +140,24 @@ def fetch_x509_bundles(self) -> X509BundleSet:
...
-        self, audiences: List[str], subject: Optional[str] = None
+        self, audiences: Set[str], subject: Optional[SpiffeId] = None
     ) -> JwtSvid:
         """Fetches a SPIFFE JWT-SVID.
 
         Args:
-            audiences: List of audiences for the JWT SVID.
-            subject: SPIFFE ID subject for the JWT.
+            audiences: Set of audiences for the JWT SVID.
+            subject: SpiffeID subject for the JWT.
```
Comment (maxlambrecht) [inline src/pyspiffe/workloadapi/default_workload_api_client.py:149]: nit: `SpiffeId`

File: src/pyspiffe/workloadapi/default_workload_api_client.py [hunk 5]
```diff
@@ -139,24 +140,24 @@ def fetch_x509_bundles(self) -> X509BundleSet:
...
 
         Returns:
             JwtSvid: Instance of JwtSvid object.
         Raises:
             ValueError: In case audience is empty.
-            FetchJwtSvidError: In case there is an error in fetching the JWTSVID from agent.
+            FetchJwtSvidError: In case there is an error in fetching the JWTSVID from Workload API.
```
Comment (maxlambrecht) [inline src/pyspiffe/workloadapi/default_workload_api_client.py:155]: very nit: `from the Workload API.`

File: src/pyspiffe/workloadapi/default_workload_api_client.py (hunk 4 above)
Comment (Andres-GC) [reply src/pyspiffe/workloadapi/default_workload_api_client.py:149]: For documentation purposes I think we should follow the same approach used on the [SPIFFE website](https://spiffe.io/) and the [GitHub repo](https://github.com/spiffe/spiffe). In both places it is used as `SPIFFE ID`.
What do you think?

**Review by Andres-GC — DISMISSED** [review]
LGTM.

File: src/pyspiffe/workloadapi/default_workload_api_client.py (hunk 4 above)
Comment (maxlambrecht) [reply src/pyspiffe/workloadapi/default_workload_api_client.py:149]: makes sense, `SPIFFE ID` is fine

Comment (maxlambrecht) [reply src/pyspiffe/workloadapi/default_workload_api_client.py:149]: As a separate task it is worth looking for other places where we use the type name `SpiffeId` in the documentation, to keep consistency.

Comment (glaucimar) [reply src/pyspiffe/workloadapi/default_workload_api_client.py:149]: Fine, I've changed to SPIFFE ID where it concerns to this PR.

File: src/pyspiffe/workloadapi/default_workload_api_client.py (hunk 5 above)
Comment (glaucimar) [reply src/pyspiffe/workloadapi/default_workload_api_client.py:155]: Done.

**Review by Andres-GC — APPROVED** [review]

**Review by maxlambrecht — APPROVED** [review]
💯

**Pull request merged**